File: app/src/main/java/eu/toma/dev/playground/warcraft/api/WoWApi.java
package eu.toma.dev.playground.warcraft.api;

import java.util.List;

import eu.toma.dev.playground.warcraft.model.boss.Boss;
import eu.toma.dev.playground.warcraft.mounts.ResponseGetMounts;
import eu.toma.dev.playground.warcraft.talent.ResponseTalents;
import io.reactivex.Observable;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
* https://dev.battle.net/io-docs
*/
public interface WoWApi
{
    String apikey = "apikey";

    @GET("wow/boss/")
    Observable<List<Boss>> getBosses(@Query(apikey) String value);

    @GET("wow/boss/{bossId}")
    Observable<Boss> getBoss(@Query(apikey) String value, @Path("bossId") int bossId);

    @GET("wow/mount/")
    Observable<Response<ResponseGetMounts>> getMounts(@Query(apikey) String value);

    @GET
    Observable<ResponseTalents> getTalents(@Query(apikey)  String value);
}

File: app/src/main/kotlin/warcraft/api/WoWApi.kt
package warcraft.api

import eu.toma.dev.playground.warcraft.talent.ResponseTalents
import io.reactivex.Observable
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query
import warcraft.model.boss.Boss
import warcraft.mounts.ResponseGetMounts

/**
 * https://dev.battle.net/io-docs
 */
interface WoWApi {

//    @GET("wow/boss/")
//    fun getBosses(@Query(apikey) value: String): Observable<List<Boss>>
//
//    @GET("wow/boss/{bossId}")
//    fun getBoss(@Query(apikey) value: String,
//                @Path("bossId") bossId: Int): Observable<Boss>

    @GET("wow/mount/")
    fun getMounts(@Query(apikey) value: String): Observable<Response<ResponseGetMounts>>

    companion object {
        const val apikey = "apikey"
    }
}
